li-apache-kafka-clients/src/main/java/com/linkedin/kafka/clients/utils/CompositeCollection.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2019 LinkedIn Corp. Licensed under the BSD 2-Clause License (the "License").  See License in the project root for license information.
+ */
+
+package com.linkedin.kafka.clients.utils;
+
+import java.util.Collection;
+import java.util.Iterator;
+
+/**
+ * quick and simple unmodifiable implementation of a collection on top of a pair of other collections.
+ * @param <T> value type
+ */
+public class CompositeCollection<T> implements Collection<T> {
+  private final Collection<T> a;
+  private final Collection<T> b;
+
+  public CompositeCollection(Collection<T> a, Collection<T> b) {
+    if (a == null || b == null) {
+      throw new IllegalArgumentException("arguments must not be null");
+    }
+    this.a = a;
+    this.b = b;
+  }
+
+  @Override
+  public int size() {
+    return a.size() + b.size();
```

Comment:
check null for a and b? Here and in other places.

Reply:
i do, at the constructor